pymc/aesaraf.py

```diff
@@ -893,11 +897,58 @@ def local_check_parameter_to_ninf_switch(fgraph, node):
 )
 
 
+def find_rng_nodes(variables: Iterable[Optional[TensorVariable]]):
+    """Return RNG variables in a graph"""
+    return [
+        node
+        for node in graph_inputs(variables)
+        if isinstance(
+            node,
+            (
+                at.random.var.RandomStateSharedVariable,
+                at.random.var.RandomGeneratorSharedVariable,
+            ),
+        )
+    ]
+
+
+def reseed_rngs(rngs: Iterable[SharedVariable], seed: "RandomSeed") -> None:
+    """Create a new set of RandomState/Generator for each rng based on a seed"""
+    bit_generators = [
+        np.random.PCG64(sub_seed) for sub_seed in np.random.SeedSequence(seed).spawn(len(rngs))
```

Comment:
Since we're at this, why don't we switch to [PCG-64 DXSM](https://numpy.org/doc/stable/reference/random/bit_generators/pcg64dxsm.html) by default. [Numpy says](https://numpy.org/doc/stable/reference/random/upgrading-pcg64.html#upgrading-pcg64) that the default PCG-64 has some statistical weaknesses in very parallel settings.

Reply:
I checked that but that bit_generator was not available in my NumPy version (1.20.3). I added an explicit test with information about the decision that will have to be done if NumPy switches their default. Until they do, I think we can keep as is: https://github.com/pymc-devs/pymc/blob/8a3cb40504e06de5a51ad4c38d84aa3771fbd3f8/pymc/tests/test_aesaraf.py#L515-L521